OMCompiler/Compiler/BackEnd/RemoveSimpleEquations.mo

```diff
@@ -192,6 +194,74 @@ algorithm
   end if;
 end removeVerySimpleEquations;
 
+public function fixAliasVarsVariablity
+" This is a workaround for #5673
+  This module traverses all alias variables and double-checks if the referenced alias variables variablitly
+  is parameter or not and changes variablilty of alias variable to BackendDAE.PARAM()."
+  input BackendDAE.BackendDAE inDAE;
+  output BackendDAE.BackendDAE outDAE;
+protected
+  BackendDAE.Variables aliasVars,systvars,globalKnownVars;
+  DAE.Exp binding;
+  DAE.ComponentRef cr;
+  Boolean paramOrConst=false,const=false;
+  BackendDAE.Shared shared;
+  list<BackendDAE.Var> referencevar,knownVarList={},aliasVarList={};
+  BackendDAE.Var tempvar;
+algorithm
+  aliasVars := BackendDAEUtil.getAliasVars(inDAE);
+  systvars := BackendVariable.listVar(BackendVariable.equationSystemsVarsLst(inDAE.eqs));
+  //BackendDump.dumpVariables(aliasVars,"aliasVariable-Actual");
+  for var in BackendVariable.varList(aliasVars) loop
+    binding := BackendVariable.varBindExp(var);
+    {cr} := Expression.getAllCrefs(binding);
+    // look up for the reference variable variablity
+    referencevar:= getVarsHelper(cr,systvars); // first check in EqSystem.OrderedVars
+    //check in shared.globalKnownVars if not found in orderedVars
+    if(listEmpty(referencevar)) then
+      referencevar := getVarsHelper(cr,inDAE.shared.globalKnownVars);
+    end if;
+    // check list of referencevariable either PARAM() or CONST()
+    if (not listEmpty(referencevar)) then
+      paramOrConst := List.mapAllValueBool(referencevar, BackendVariable.isParamOrConstant, true);
+      const := List.mapAllValueBool(referencevar, BackendVariable.isConst, true);
+    end if;
+
+    if (paramOrConst) then
+      // remove the variable from AliasVarList and add it to GlobalknownVarlist after changing it to PARAM() and fixed=true
+      tempvar := BackendVariable.setVarKind(var,BackendDAE.PARAM());
+      knownVarList := BackendVariable.setVarFixed(tempvar,true) :: knownVarList;
+      //BackendDump.dumpVarList(referencevar,"matchedparamvar");
+    elseif (const) then
+      tempvar := BackendVariable.setVarKind(var,BackendDAE.CONST());
+      knownVarList := BackendVariable.setVarFixed(tempvar,true) :: knownVarList;
```

Comment:
Not sure, but don't these two cases need to be changed in order?

Reply:
@lochel I have made the changes this will make parameters and constants correctly, we will see how it goes